### ISSUE TITLE
Order of attributes breaks tag parsing

### ISSUE THREAD

**Issue by andrelaszlo** [AdamNiederer/vue-mode#81]
I noticed that the highlighting didn't work for a few of my components. For any style with this tag, the style tag didn't use the correct mode:

```
<style scoped lang="scss">
  // My style
</style>
```
I started reading `vue-mode.el` and found the regex that parses the tags, and it seems that scoped has to appear after the lang attribute, so the below will work:

```
<style lang="scss" scoped>
  // My style
</style>
```

I didn't expect the order of attributes to matter for a html-like language (it's a bit unclear to me exactly which standard the Vue component syntax is based on).

Similarly, if I put `module` before `scoped` it doesn't parse correctly, but if I reverse the order, ie `scoped module`, it parses correctly again.

**Comment by DougBeney** [comment]
Just came here to report the same issue. Same behavior.

**Comment by DougBeney** [comment]
Spent some time trying to fix it and I believe this will work.

```
(defconst vue--front-tag-lang-regex
  (concat "<%s"                              ; The tag name
          "\\(?: +scoped\\)?"                ; The optional "scoped" attribute
          "\\(?: +module\\)?"                ; The optional "module" attribute
          "\\(?: +\\w+=[\"'].*?[\"'] *?\\)*" ; Any optional key-value pairs like type="foo/bar"
          "\\(?: +scoped\\)?"                ; The optional "scoped" attribute
          "\\(?: +module\\)?"                ; The optional "module" attribute
          " +lang=[\"']%s[\"']"              ; The language specifier
          "\\(?: +\\w+=[\"'].*?[\"'] *?\\)*" ; More optional key-value pairs
          "\\(?: +scoped\\)?"                ; The optional "scoped" attribute
          "\\(?: +module\\)?"                ; The optional "module" attribute
          " *>\n")                           ; The end of the tag
  "A regular expression for the starting tags of template areas with languages.
To be formatted with the tag name, and the language.")
```

@AdamNiederer How do you set up a development environment to work on this package? I was having trouble with that so I wasn't able to test this code.

**Comment by AdamNiederer** [comment]
> How do you set up a development environment to work on this package? I was having trouble with that so I wasn't able to test this code.

You may have success just using `re-builder` on you affected pieces of code. I think the proposed regex can be significantly simplified, though, with a well-placed alternation or two. I'll see what I can do with it this weekend.

**Comment by AdamNiederer** [comment]
This should theoretically be resolved in e5faa5767308dcd87139813957eabba62b7caf19.
